sparfa_server/api.py

```diff
@@ -0,0 +1,173 @@
+
+blueprint = Blueprint('API', __name__)
+
+
+@blueprint.route('/fetch_ecosystem_matrices', methods=['POST'])
+def show():
+    if 'Biglearn-Sparfa-Token' not in request.headers:
+        raise Unauthorized('Request missing the Biglearn-Sparfa-Token header')
+
+    if request.headers['Biglearn-Sparfa-Token'] != BIGLEARN_SPARFA_TOKEN:
+        raise Unauthorized('Invalid Biglearn-Sparfa-Token header')
+
+    request_data = request.get_json(force=True)
+
+    FETCH_ECOSYSTEM_MATRICES_VALIDATOR(request_data)
+
+    ecosystem_matrix_requests = request_data['ecosystem_matrix_requests']
+
+    ecosystem_matrix_uuids = []
+    student_uuids_by_request_uuid = {}
+    responded_before_by_request_uuid = {}
+    for em_request in ecosystem_matrix_requests:
+        ecosystem_matrix_uuids.append(em_request['ecosystem_matrix_uuid'])
+        request_uuid = em_request['request_uuid']
+        student_uuids_by_request_uuid[request_uuid] = em_request.get('student_uuids')
+        responded_before_by_request_uuid[request_uuid] = em_request.get('responded_before')
+
+    with transaction() as session:
+        ecosystem_matrices_by_uuid = {
+            ecosystem_matrix.uuid: ecosystem_matrix
+            for ecosystem_matrix in session.query(EcosystemMatrix).filter(
+                EcosystemMatrix.uuid.in_(ecosystem_matrix_uuids)
+            ).all()
+        }
+
+        ecosystem_matrix_responses = []
+        for ecosystem_matrix_request in ecosystem_matrix_requests:
+            request_uuid = ecosystem_matrix_request['request_uuid']
+            ecosystem_matrix_uuid = ecosystem_matrix_request['ecosystem_matrix_uuid']
+            student_uuids = student_uuids_by_request_uuid[request_uuid]
+            ecosystem_matrix_response = {
+                'request_uuid': request_uuid,
+                'ecosystem_matrix_uuid': ecosystem_matrix_uuid,
+                'responded_before': ecosystem_matrix_request.get('responded_before'),
+                'ecosystem_uuid': None,
+                'L_ids': [] if student_uuids is None else student_uuids,
+                'Q_ids': [],
+                'C_ids': [],
+                'd_data': [],
+                'W_data': [],
+                'W_row': [],
+                'W_col': [],
+                'H_mask_data': [],
+                'H_mask_row': [],
+                'H_mask_col': [],
+                'G_data': [],
+                'G_row': [],
+                'G_col': [],
+                'G_mask_data': [],
+                'G_mask_row': [],
+                'G_mask_col': [],
+                'U_data': [],
+                'U_row': [],
+                'U_col': [],
+                'superseded_at': None
+            }
+
+            ecosystem_matrix = ecosystem_matrices_by_uuid.get(ecosystem_matrix_uuid)
+            if ecosystem_matrix:
+                ecosystem_matrix_response.update({
+                    key: value if not hasattr(value, 'isoformat') else value.isoformat() + 'Z'
```

Comment:
it's usually better to use just `if` vs `if not`, could this could be flipped as:
```python
key: value.isoformat() if hasattr(value, 'isoformat') else value
```